### DISCUSSION TITLE
Is there a WEBSOCKETS_URL?

### DISCUSSION THREAD

**Discussion by itskingori** [outline/outline#3067] [Self Hosting]
> The websocket server is used to communicate with the frontend, it can be ran on the same box as the web server or separately.

That's according [to the `SERVICES.md` file](https://github.com/outline/outline/blob/d643c9453e8c2ecf3bbc8aa249a81b7f2aaa41b2/docs/SERVICES.md). But how? 🤔 Same doc states that you can use the `COLLABORATION_URL` to set the collaboration server but I can't find an equivalent for the websocket server. Checked the code as well, and can't find anything ...

https://github.com/outline/outline/blob/d643c9453e8c2ecf3bbc8aa249a81b7f2aaa41b2/shared/types.ts#L6-L19

https://github.com/outline/outline/blob/c8a67c374f972cf7af26b9f5e012eebde645c93f/server/services/websockets.ts#L17-L24

**Comment by tommoor** [comment]
@itskingori I believe the document is mistaken here, it may have been possible in the past but as you found the url is hardcoded to the same as the webserver right now. Running it separate is not necessary for self-hosting irregardless.

**Reply by itskingori** [reply]
Right, noted. We have two load balancers one that supports websockets (and doesn't support http) and another that supports http (but not websockets). So we wanted to run web separately from websockets/collaboration (which are both websocket services).
